### PR TITLE
Add Textarea validation state and message.

### PR DESCRIPTION
**Pull request by marioharper**
## Description

Added validation state and message similar to how the `Field` component currenlty works / looks.
Also removed `TextareaWithoutLabel`. Now `Textarea` just adds a label if there is a `label` property passed to it. 

## Related Issue

#194 

## Motivation and Context

Keeps form validation consistent throughout The Knot. 

## Types of changes

- [x] Documentation
- [ ] Bug fix (non-breaking change which fixes an issue. you didn't modify existing tests)
- [ ] New feature (non-breaking change which adds functionality. you added at least one new test)
- [x] Breaking change (fix or feature that would cause existing functionality to change. you had to modify existing tests.)
- [ ] Does this introduce a new pattern/package?

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/xogroup/union/blob/master/.github/CONTRIBUTING.md) document.
- [x] I have updated the documentation as needed.
- [x] I have added tests to cover my changes.

## Reviewer Checklist

- [ ] Is change documented in a release draft? (Only if there are plans to merge PR)

### CODE REVIEW

File: packages/@xo-union/tk-component-fields/src/components/Textarea/index.jsx [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
-  label = labelize(name),
+  label,
+  state = 'neutral',
+  validationMessage,
   id = fieldId(name),
   ...props
 }) {
+  const inputClass = classMap[state];
+
+  if (!inputClass) {
```
Comment (GeorgeTaveras1231) [inline packages/@xo-union/tk-component-fields/src/components/Textarea/index.jsx]: should be named `textareaClass`.

File: packages/@xo-union/tk-component-fields/src/components/Textarea/index.jsx [hunk 2]
```diff
@@ -1,18 +1,33 @@
...
+const classMap = {
+  neutral: FieldsCss.textarea,
+  invalid: FieldsCss.invalidTextarea,
+  valid: FieldsCss.validTextarea
+};
 
 export default function Textarea({
   name,
-  label = labelize(name),
+  label,
```
Comment (GeorgeTaveras1231) [inline packages/@xo-union/tk-component-fields/src/components/Textarea/index.jsx]: I think the preferred behavior would be for the label to be assumed when it is not provided. So

```jsx
<Textarea name="my-field" />
```

Should have a label of `"My field"`. This is helpful in most cases, and maintains parity with the current implementation of `<Field />`.

If a consumer wants to opt out of that behaviour, they can do:

```jsx
<Textarea name="my-field" label={false} />
```

File: packages/@xo-union/tk-component-fields/src/components/Textarea/index.jsx [hunk 3]
```diff
@@ -1,18 +1,33 @@
...
+  state = 'neutral',
+  validationMessage,
   id = fieldId(name),
   ...props
 }) {
+  const inputClass = classMap[state];
```
Comment (GeorgeTaveras1231) [inline packages/@xo-union/tk-component-fields/src/components/Textarea/index.jsx]: We should have unit tests for this logic: Which class is applied based on the given `state`

Comment (marioharper) [inline packages/@xo-union/tk-component-fields/src/components/Textarea/index.jsx]: These unit tests cover that https://github.com/xogroup/union/pull/196/files#diff-d2374d2fb8d6cca58e5d25988675bea1R30.

**Comment by marioharper** [comment]
@GeorgeTaveras1231 requested corrections made

**Comment by GeorgeTaveras1231** [comment]
Thanks @marioharper 👍. We decided to merge this PR before #186 since it is much simpler. Sorry for the delay, and thanks again for the contribution.

**Comment by GeorgeTaveras1231** [comment]
Release `@xo-union/tk-component-fields@1.0.0`

**Comment by marioharper** [comment]
Awesome, thanks @GeorgeTaveras1231. Looks like I introduced some weirdness though... Looks like a z-index issue. Noticed this on the top demo at http://docs.union.theknot.com/pattern-library/core-components/form-elements. 

screen shot 2017-07-19 at 10 56 44 am

I'll open an issue to track this.

**Pull request merged**